src/pages/books/Book.jsx

```diff
@@ -33,6 +34,27 @@ const Book = () => {
     fetchData();
   }, [id, location.state, navigate]);
 
+  const handleAddToCart = async () => {
+    if (!isLoggedIn) {
+      navigate('/sign_in'); // Redirect to login if the user is not logged in
+      return;
+    }
+
+    setIsAddingToCart(true);
+    try {
+      await addToCart(
+        { 'Content-Type': 'application/json' },
+        { bookId: id, price: bookData.price },
+        token
+      );
+      navigate('/cart'); // Redirect to the cart page after adding to cart
```

Comment:
The comments explaining the navigate calls might be redundant since the code is self-explanatory. Removing these comments could make the code cleaner and more concise.